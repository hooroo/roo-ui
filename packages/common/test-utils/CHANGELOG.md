# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
