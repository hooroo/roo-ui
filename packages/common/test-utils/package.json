{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
