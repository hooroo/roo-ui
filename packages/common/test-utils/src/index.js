export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
