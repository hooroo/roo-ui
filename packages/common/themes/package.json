{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
