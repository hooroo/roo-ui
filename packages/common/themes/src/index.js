import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
