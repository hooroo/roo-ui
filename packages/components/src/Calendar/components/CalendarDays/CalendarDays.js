import React from 'react';
import PropTypes from 'prop-types';
import { themeGet } from 'styled-system';
import { css } from 'styled-components';
import { darken } from 'polished';

import { Flex, NakedButton } from '../../../';

const Wrapper = Flex.extend`
  flex: 1 1 auto;
  width: calc(100% / 7);
  justify-content: center;
  margin: 0 -1px -1px 0;
  border: ${themeGet('borders.1')} ${themeGet('colors.grey.3')};

  &:after {
    content: '';
    display: block;
    padding-bottom: 100%;
  }
`;

const Button = NakedButton.extend`
  color: ${themeGet('colors.grey.0')};
  padding: 0;
  width: 100%;
  border: ${themeGet('borders.2')} transparent;


  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: not-allowed;
    background-color: ${themeGet('colors.grey.2')};
  }

  ${props => props.selectable &&
    css`
      background-color: ${themeGet('colors.white')};

      &:hover,
      &:focus {
        border-color: ${themeGet('colors.brand.secondary')};
      }

      &:active {
        background-color: ${themeGet('colors.ui.infoBackground')};
      }
  `};

  ${props => props.selected &&
    css`
      background-color: ${themeGet('colors.ui.infoBackground')};
      border-color: ${themeGet('colors.brand.secondary')};

      &:hover,
      &:focus {
        background-color: ${themeGet('colors.white')};
      }
  `};

  ${props => !props.selectable && !props.disabled &&
    css`
      background-color: ${themeGet('colors.grey.2')};

      &:hover,
      &:focus {
        background-color: ${darken(0.1, themeGet('colors.grey.2')(props))};
      }
  `};
`;

Button.defaultProps = {
  ...NakedButton.defaultProps,
  disabled: PropTypes.bool,
  blacklist: [...Object.keys(NakedButton.propTypes), 'selectable'],
};

export const CalendarDays = Flex.extend`
  flex-wrap: wrap;
  margin-bottom: 1px;
  margin-right: 1px;
`;

export const CalendarEmptyDay = Wrapper.withComponent('div').extend`
  border-color: transparent;
`;

CalendarEmptyDay.displayName = 'CalendarEmptyDay';

export const CalendarDay = ({ children, ...rest }) => (
  <Wrapper>
    <Button {...rest}>
      {children}
    </Button>
  </Wrapper>
);

CalendarDay.propTypes = {
  children: PropTypes.node.isRequired,
};
