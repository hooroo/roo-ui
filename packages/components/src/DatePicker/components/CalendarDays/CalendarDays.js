import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';

import { Flex } from '../../../';
import { CalendarDay, CalendarEmptyDay } from '../CalendarDay';

const CalendarDaysWrapper = styled(Flex)`
  flex-wrap: wrap;
  margin-bottom: 2px;
  margin-right: 2px;
`;

const getCustomDateProps = (disabledDates, interactiveDisabledDates, isInRange, day) => {
  const isDisabled = disabledDates
    .some(disabledDate => isEqual(String(disabledDate), String(day.date)));

  const props = {
    selected: day.selected || (interactiveDisabledDates && day.selectable),
    highlighted: isInRange(day.date),
    selectable: day.selectable,
  };

  if (isDisabled) {
    props.selectable = false;
    props.highlighted = false;
    props.disabled = !interactiveDisabledDates;
  }

  return props;
};

const CalendarDays = ({
  weeks, month, year, getDateProps, disabledDates, interactiveDisabledDates,
  onMouseEnterOfDay, isInRange,
}) => (
  <CalendarDaysWrapper>
    {weeks.map(week =>
        week.map((day, index) => {
          if (!day) {
            return (
              <CalendarEmptyDay key={`${year}${month}${index}`} /> // eslint-disable-line react/no-array-index-key
            );
          }

          return (
            <CalendarDay
              key={`${year}${month}${index}`} // eslint-disable-line react/no-array-index-key
              {...getDateProps({
                dateObj: day,
                onMouseEnter: () => onMouseEnterOfDay(day),
              })}
              {...getCustomDateProps(disabledDates, interactiveDisabledDates, isInRange, day)}
            >
              {day.date.getDate()}
            </CalendarDay>
          );
        }))}
  </CalendarDaysWrapper>
);

CalendarDays.defaultProps = {
  disabledDates: [],
  interactiveDisabledDates: false,
  isInRange: () => {},
  onMouseEnterOfDay: () => {},
};

CalendarDays.propTypes = {
  month: PropTypes.number.isRequired,
  year: PropTypes.number.isRequired,
  weeks: PropTypes.arrayOf(PropTypes.array).isRequired,
  getDateProps: PropTypes.func.isRequired,
  disabledDates: PropTypes.arrayOf(PropTypes.instanceOf(Date)),
  interactiveDisabledDates: PropTypes.bool,
  onMouseEnterOfDay: PropTypes.func,
  isInRange: PropTypes.func,
};

export default CalendarDays;
