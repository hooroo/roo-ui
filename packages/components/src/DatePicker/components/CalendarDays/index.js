export { CalendarDay, CalendarEmptyDay, CalendarDays } from './CalendarDays';
