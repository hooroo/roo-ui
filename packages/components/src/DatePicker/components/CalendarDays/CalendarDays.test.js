import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import CalendarDays from '.';

describe('<CalendarDays />', () => {
  it('renders correctly', () => {
    const component = <CalendarDays selected selectable>Day</CalendarDays>;
    expect(shallowWithTheme(component, theme)).toMatchSnapshot();
  });
});
