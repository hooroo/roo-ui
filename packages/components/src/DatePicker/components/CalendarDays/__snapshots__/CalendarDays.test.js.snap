// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarDays /> renders correctly 1`] = `
<CalendarDays__CalendarDaysWrapper
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "width",
      "fontSize",
      "color",
      "bg",
      "fontFamily",
      "textAlign",
      "lineHeight",
      "fontWeight",
      "fontStyle",
      "letterSpacing",
      "display",
      "maxWidth",
      "minWidth",
      "height",
      "maxHeight",
      "minHeight",
      "size",
      "ratio",
      "verticalAlign",
      "alignItems",
      "alignContent",
      "justifyItems",
      "justifyContent",
      "flexWrap",
      "flexBasis",
      "flexDirection",
      "flex",
      "justifySelf",
      "alignSelf",
      "order",
      "gridGap",
      "gridColumnGap",
      "gridRowGap",
      "gridColumn",
      "gridRow",
      "gridAutoFlow",
      "gridAutoColumns",
      "gridAutoRows",
      "gridTemplateColumns",
      "gridTemplateRows",
      "gridTemplateAreas",
      "gridArea",
      "border",
      "borderTop",
      "borderRight",
      "borderBottom",
      "borderLeft",
      "borderColor",
      "borderRadius",
      "boxShadow",
      "opacity",
      "overflow",
      "background",
      "backgroundImage",
      "backgroundPosition",
      "backgroundRepeat",
      "backgroundSize",
      "position",
      "zIndex",
      "top",
      "right",
      "bottom",
      "left",
      "textStyle",
      "colors",
      "variant",
      "theme",
    ]
  }
>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201870"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    1
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201871"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    2
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201872"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    3
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201873"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    4
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201874"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    5
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201875"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    6
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201876"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    8
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201870"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    8
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201871"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    9
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201872"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    10
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201873"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    11
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201874"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    12
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201875"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    13
  </CalendarDay>
  <CalendarDay
    _isMockFunction={true}
    disabled={false}
    getMockImplementation={[Function]}
    getMockName={[Function]}
    highlighted={false}
    key="201876"
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
    selectable={true}
    selected={false}
  >
    15
  </CalendarDay>
</CalendarDays__CalendarDaysWrapper>
`;
